### ISSUE TITLE
New using-engine removes using statements if referring to package

### ISSUE THREAD

**Issue by Ton-Pe** [omecra/de.omecra.oedt.updatesite#105]
OEDT version 9.3.5. New using-engine activated.

If using statements are referring to package (using * wildcard). Saving the file (applying correct using statements) removes all those using statements. If using statements are referring to actual class, those will remain. 

```
USING OpenEdge.Net.HTTP.HttpRequest FROM PROPATH.
USING OpenEdge.Web.* FROM PROPATH.
USING Progress.Lang.*  FROM PROPATH.

BLOCK-LEVEL ON ERROR UNDO, THROW.

CLASS ABL.Classes.TestUsing:

    METHOD PUBLIC VOID Test1():

        DEFINE VARIABLE xWebRequest AS WebRequest NO-UNDO.
        DEFINE VARIABLE xHttpRequest AS HttpRequest NO-UNDO.

        CATCH xError AS AppError :
        END CATCH.

    END.
END CLASS.
```
When Correct using is applied, code will be:
```
USING OpenEdge.Net.HTTP.HttpRequest FROM PROPATH.

BLOCK-LEVEL ON ERROR UNDO, THROW.

CLASS ABL.Classes.TestUsing:

    METHOD PUBLIC VOID Test1():

        DEFINE VARIABLE xWebRequest AS WebRequest NO-UNDO.
        DEFINE VARIABLE xHttpRequest AS HttpRequest NO-UNDO.

        CATCH xError AS AppError :
        END CATCH.

    END.
END CLASS.
```

![image](https://user-images.githubusercontent.com/25682299/95312028-14624300-0897-11eb-871c-1c5904c1568c.png)

Well, OEDT is set to add whole package, not a wildcard, into using statement when adding using statement. But this selection does not affect on removing the statements when saving. We have lot's of generated code that has wildcard references, although all manually added using statements need to have full class reference.

**Comment by lneumeier** [comment]
The removal of 
```
USING OpenEdge.Web.* FROM PROPATH.
USING Progress.Lang.*  FROM PROPATH.
```
is expected, since "use Wildcard (.*)" is not activated. You could deactivate "Remove unused using statements" add a workaround.

Since "Add missing using statements" is activated, not adding WebRequest as a using statement is a bug.

**Comment by Ton-Pe** [comment]
OK.

Activating "Use Wildcard (.*)" still results removal of
```
USING OpenEdge.Web.* FROM PROPATH.
USING Progress.Lang.*  FROM PROPATH.
```
if "Remove unused using statements" is activated. But it changes the `USING OpenEdge.Net.HTTP.HttpRequest FROM PROPATH.` into `USING OpenEdge.Net.HTTP.* FROM PROPATH.`

![image](https://user-images.githubusercontent.com/25682299/95416709-f69ae980-093b-11eb-94ed-74978b844cc9.png)

```
USING OpenEdge.Web.* FROM PROPATH.
USING Progress.Lang.*  FROM PROPATH.
```
are removed even if I deactivate the "Remove unused using statements". (And use wildcard is deactivated)

![image](https://user-images.githubusercontent.com/25682299/95416951-850f6b00-093c-11eb-93b7-a2cea6bc5229.png)

**Comment by Ton-Pe** [comment]
> Does this happened too, if you just save the file?
What do you mean by this? I have "Execute on save" activated so this is always executed on save. If I deactivate execution on save, then just saving the file works OK. It won't change any of those using statements, as expected.

**Comment by lneumeier** [comment]
Fixed in 9.3.6.
